### ISSUE TITLE
Focus not logical when error committed using the error summary component

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-frontend#1398]
This issue is from a May 2019 external accessibility audit report.

**WCAG Reference**: [2.4.3 Focus Order (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html)
**Issue ID**: DAC_Issue14
**URL**: http://govuk-frontend-v3.herokuapp.com/full-page-examples/passport-details#expiry-month
**Page**: Passport details

## Summary

When an error had occurred, focus was taken to the second date field which was disorientating for some users

## Screen Shot
![](https://user-images.githubusercontent.com/2445413/58707492-051edb80-83ad-11e9-86db-842abc8e3a06.png)

When the form fields were left empty and the form submitted, the users focus was jumped to the second date entry field. This was confusing to users as it was not clear why this field had received focus.

## Current Code Ref(s)
```html
<input class="govuk-input govuk-date-input__input govuk-input--width-2 govuk-input--error" id="expiry-month" name="expiry-month" type="number" pattern="[0-9]*">
```

## Screen reader comments
> “When I committed an error by leaving all of the form fields empty, I found that my focus was taken to the ‘Month’ field and the heading ‘There is a problem’ as expected. I was not sure whether I was supposed to enter the ‘Day’ field as this had been missed.”

## Keyboard only comments:
> “I could not understand why my focus was taken to the second date field when I hadn’t entered any data into any of the form fields. I had to re-navigate the page to get to the list of error links at the top of the form which was time consuming for me.”

## Solution
Ensure that when the user commits an error on the page, focus is sent to the error summary heading. Once the error link is selected, focus must be placed on the first field in a group.

**Comment by nickcolley** [comment]
I have assigned priority high since after an initial discussion with the team.

Since when the error summary links are clicked they leave the anchor id for the input in the URL, if you submit the form again you will skip to the previously errored input.

In [GOV.UK Elements this behaviour was prevented](https://github.com/alphagov/govuk_elements/blob/ef0baed92fabbbfdc0f6ea09d7dd76bffa994bf8/assets/javascripts/application.js#L29) but we did not introduce this in GOV.UK Frontend as we were not sure why this was done.

We could either

1. Write guidance to ensure users do not use `<form>` elements without an explicit 'action'
2. Re-introduce the behaviour in GOV.UK Elements which stops the anchor ID from being added to the URL when interacting with the error summary component.

I would recommend doing 2. since 1. feels easy to miss for our users.
